packages/storage-plugin/src/symbols.ts

```diff
@@ -44,14 +44,21 @@ export interface NgxsStoragePluginOptions {
   }[];
 
   /**
-   * Serailizer for the object before its pushed into the engine.
+   * Serializer for the object before its pushed into the engine.
+   *
+   * @param obj {any} value of currently processed state slice
+   * @param key {string} key of currently processed state slice
    */
-  serialize?(obj: any);
+  serialize?(obj: any, key?: any);
 
   /**
    * Deserializer for the object before its pulled out of the engine.
+   *
+   * @param obj {any} value of currently processed state slice
+   * @param key {string} key of currently processed state slice
+   * @param state {any} previous value of currently processed state slice
    */
-  deserialize?(obj: any);
+  deserialize?(obj: any, key?: any, state?: any);
```

Comment:
Unfortunately switching the key to any is a bit of a hack to avoid the breaking change. I would prefer that it was the `string` type. I really need to understand the need for this feature though...